clang/tools/driver/cc1_main.cpp

```diff
@@ -216,12 +222,30 @@ int cc1_main(ArrayRef<const char *> Argv, const char *Argv0, void *MainAddr) {
     return 1;
 
   // Execute the frontend actions.
-  Success = ExecuteCompilerInvocation(Clang.get());
+  {
+    llvm::TimeTraceScope scope("ExecuteCompiler", "");
+    Success = ExecuteCompilerInvocation(Clang.get());
+  }
 
   // If any timers were active but haven't been destroyed yet, print their
   // results now.  This happens in -disable-free mode.
   llvm::TimerGroup::printAll(llvm::errs());
 
+  if (llvm::TimeTraceProfilerEnabled()) {
+    SmallString<128> Path(Clang->getFrontendOpts().OutputFile);
```

Comment:
When using `clang++ -ftime-trace main.cpp -o main` on Windows this will generate a file in `%TEMP%\main-xxx.json` is that the intended behavior?

Reply:
I think so -- basically this produces a .json file for each object file.

So where that is placed is fairly natural if you're just invoking clang to compile one file (with `-c` argument). However if you're invoking it to both compile and link, then it's a good question at where should the json files be placed (since there can be more than one!), or whether they should be merged somehow, or should just be an error. Today that's not very intuitive as you noticed.

Reply:
My use case is that I'd like to profile compile times of individual files in my Visual Studio solution with the help Clang Power Tools. Since it uses the `-fsyntax-only` flag the output path will be an empty string, so it outputs a `.json` file. For my use case I can work around it, but perhaps it's good to consider these questions.

Thanks a lot for your work, I needed this for a long time!

Reply:
Ah. My use case was "I know which files are slow, I want to look what's inside of each file". The "which files are slow" in my case is already answered by the build system that we use, since it just invokes clang for each.

But yeah "some sort of output" when one clang invocation processes multiple input files would be useful. Will have to think about it.